File: web/ops/discord_utils.js
import fetch from 'node-fetch';
import logger from './logger.js';

export async function sendOpsMessage(message, options = {}) {
  const webhookUrl = options.webhookUrl || process.env.DISCORD_WEBHOOK_OPS;
  if (!webhookUrl) {
    const error = 'DISCORD_WEBHOOK_OPS is not set';
    logger.warn({ error }, 'sendOpsMessage skipped');
    return { ok: false, error };
  }
  const content = typeof message === 'string' ? message : JSON.stringify(message);
  const body = { content };
  const resp = await fetch(webhookUrl, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(body)
  });
  const ok = resp.ok;
  if (!ok) {
    const text = await resp.text().catch(() => '');
    logger.error({ status: resp.status, text }, 'Discord webhook failed');
    return { ok: false, status: resp.status, text };
  }
  return { ok: true };
}

File: krizzy_ops/utils/discord_utils.py
import os,aiohttp
WEBHOOK=os.getenv("DISCORD_WEBHOOK_OPS")

async def post_log(msg):
    if WEBHOOK:
        async with aiohttp.ClientSession() as s: await s.post(WEBHOOK,json={"content":msg})

async def post_error(msg):
    await post_log(f"🔴 {msg}")
